src/lib/events/index.ts

```diff
@@ -118,21 +118,30 @@ export abstract class EventEmitter<Events extends Default | {}> {
 	}
 
 	/**
-	 * Emits the event, either resuming the yeilded threads or invoking the do's chain.
+	 * Emit the event with the given data.
 	 *
 	 * @param token event to emit.
-	 * @param args of type T which are the parameters passed to the function definition.
-	 *
-	 * @returns a promise.
+	 * @param data of type T which are the parameters passed to the function definition.
 	 */
 	public emit<T extends keyof Events, S extends ArrayOrNever<Events[T]>>(
 		token: T,
-		...args: S
+		data: S,
 	): void {
 		if (token in this.events) {
 			for (const thread of this.events[token as string]) {
-				void thread.call(...(token === "_default" ? [] : args));
+				void thread.call(token === "_default" ? [] : data);
 			}
 		}
 	}
+
+	/**
+	 * Disconnects every binded callback to the event specified.
+	 *
+	 * @param token event to disconnect.
+	 */
+	public disconnect<T extends keyof Events>(token: T): void {
+		if (token in this.events) {
+			table.clear(this.events[token as string]);
+		}
+	}
```

Comment:
This is not the disconnect that was asked for 🤔

We should be able to disconnect individual EventListeners, you can rename *this one* to `clear`, and call the other one `EventListener#disconnect`.

Reply:
Changed.